drivers/gles3/shaders/scene.glsl

```diff
@@ -1216,7 +1229,21 @@ vec4 fog_process(vec3 vertex) {
 	}
 #endif // !DISABLE_LIGHT_DIRECTIONAL
 
-	float fog_amount = 1.0 - exp(min(0.0, -length(vertex) * scene_data.fog_density));
+	float fog_amount = 0.0;
+
+#ifndef DISABLE_FOG_DEPTH
+	if (scene_data.fog_mode == 1) {
```

Comment:
```suggestion
	if (scene_data.fog_mode == uint(1)) {
```
This fails to compile since it is a comparison between an int and a uint. GLSL is really strict about types

Reply:
If you follow my suggestion above, then you won't even need to make this change as the statement can just be deleted

Reply:
> If you follow my suggestion about the spec constant, then you won't even need to make this change as the statement can just be deleted

I made all the basic corrections, but this request is a little more advanced, I'm going to make a commit making just this adjustment today, so if there is any error we can easily revert it.